subprojects/gradle/buildchecks/src/main/kotlin/com/avito/android/plugin/build_param_check/UniqueRClassesTaskFactory.kt

```diff
@@ -0,0 +1,63 @@
+package com.avito.android.plugin.build_param_check
+
+import com.android.build.gradle.api.ApplicationVariant
+import com.android.build.gradle.internal.tasks.factory.dependsOn
+import com.android.build.gradle.tasks.ProcessApplicationManifest
+import com.android.build.gradle.tasks.ProcessTestManifest
+import com.avito.android.androidAppExtension
+import com.avito.android.isAndroidApp
+import com.avito.android.plugin.build_param_check.BuildChecksExtension.Check
+import com.avito.kotlin.dsl.isRoot
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.tasks.TaskProvider
+import org.gradle.kotlin.dsl.register
+
+internal class UniqueRClassesTaskFactory(
+    private val rootProject: Project,
+    private val config: Check.UniqueRClasses
+) {
+
+    init {
+        check(rootProject.isRoot()) {
+            "Project ${rootProject.path} must be root"
+        }
+    }
+
+    fun register(rootTask: TaskProvider<Task>) {
```

Comment:
Combination of className `UniqueRClassesTaskFactory`, function name `register` and param name `rootTask` is unclear for me
I don't know how to rename the factory but propose rename register to the dependsOn or something similar

And tasks creation is implementation details

Reply:
Renamed to this in #781 :

```kotlin
UniqueRClassesTaskProvider(...)
                .dependsOn(rootTask)
```